formatting/models.py

```diff
@@ -11,143 +11,248 @@
         return reverse("format:date_detail", kwargs={"pk": self.pk})
 
     class Meta:
         ordering = ("date_id",)
 
 
 class Time(PermissionsBase):
-    """Time format"""
-
-    time_id = models.AutoField("Id", primary_key=True)
-    time_format = models.CharField("Format", max_length=20)
+    """Time format.
+
+    Format string for the time column. It is used to parse the time column in the data
+    file. The format string must be compatible with the `datetime` module in Python. See
+    the [datetime documentation](https://docs.python.org/3.11/library/datetime.html#format-codes)
+    for more information on valid format codes.
+
+    Attributes:
+        date_id (AutoField): Primary key.
+        date_format (CharField): The format string for the date column in human readable
+            form, eg. `HH:MM:SS 24H`.
+        code (CharField): The code used to parse the date column, eg. `%H:%M:%S`.
+    """
+
+    time_id = models.AutoField("Id", primary_key=True, help_text="Primary key.")
+    time_format = models.CharField(
+        "Format",
+        max_length=20,
+        help_text="The format string for the date column in human readable form, eg. "
+        "`HH:MM:SS 24H`",
+    )
     code = models.CharField("Code", max_length=20)
 
-    def __str__(self):
+    def __str__(self) -> str:
+        """Return the string representation of the object."""
         return str(self.time_format)
 
-    def get_absolute_url(self):
+    def get_absolute_url(self) -> str:
+        """Get the absolute URL of the object."""
         return reverse("format:time_detail", kwargs={"pk": self.pk})
 
     class Meta:
         ordering = ("time_id",)
 
 
 class Format(PermissionsBase):
-    """Details of the data file format, combining different aspects."""
-
-    format_id = models.AutoField("format_id", primary_key=True)
+    """Details of the data file format, describing how to read the file.
+
+    It combines several properties, such as the file extension, the delimiter, the date
+    and time formats, and the column indices for the date and time columns, instructing
+    how to read the data file and parse the dates. It is mostly used to ingest data from
+    text files, like CSV.
+
+    Attributes:
+        format_id (AutoField): Primary key.
+        name (CharField): Short name of the format entry.
+        description (TextField): Description of the format.
+        extension (ForeignKey): The extension of the data file.
+        delimiter (ForeignKey): The delimiter between columns in the data file. Only
+            required for text files.
+        first_row (SmallIntegerField): First row of the data, excluding any heading.
+        footer_rows (SmallIntegerField): Number of footer rows, to be ignored at the
+            end.
+        date (ForeignKey): Format for the date column. Only required for text files.
+        date_column (SmallIntegerField): Index of the date column, starting in 1.
```

Comment:
```suggestion
        date_column (SmallIntegerField): Index of the date column, starting at 1.
```

Also, what about `first_row`, does this start at zero or 1?

Reply:
Yep, it also starts in 1.